File: backEnd/src/main/java/models/UserGroup.java
package models;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class UserGroup {

  private String groupName;
  private String icon;
  private String lastActivity;
  private boolean muted;
  private Map<String, Boolean> eventsUnseen;

  public UserGroup(final Map<String, Object> jsonMap) {
    this.setGroupName((String) jsonMap.get(Group.GROUP_NAME));
    this.setIcon((String) jsonMap.get(Group.ICON));
    this.setLastActivity((String) jsonMap.get(Group.LAST_ACTIVITY));
    this.setMuted(this.getBoolFromObject(jsonMap.get(AppSettings.MUTED)));
    this.setEventsUnseenRawMap((Map<String, Object>) jsonMap.get(User.EVENTS_UNSEEN));
  }

  public static UserGroup fromNewGroup(final Group newGroup) {
    return new UserGroup(
        newGroup.getGroupName(),
        newGroup.getIcon(),
        newGroup.getLastActivity(),
        false,
        Collections.emptyMap()
    );
  }

  public Map<String, Object> asMap() {
    final Map<String, Object> modelAsMap = new HashMap<>();
    modelAsMap.putIfAbsent(Group.GROUP_NAME, this.groupName);
    modelAsMap.putIfAbsent(Group.ICON, this.icon);
    modelAsMap.putIfAbsent(Group.LAST_ACTIVITY, this.lastActivity);
    modelAsMap.putIfAbsent(AppSettings.MUTED, this.muted);
    modelAsMap.putIfAbsent(User.EVENTS_UNSEEN, this.eventsUnseen);
    return modelAsMap;
  }

  public void setEventsUnseenRawMap(final Map<String, Object> jsonMap) {
    this.eventsUnseen = null;
    if (jsonMap != null) {
      this.eventsUnseen = new HashMap<>();
      for (String eventId : jsonMap.keySet()) {
        this.eventsUnseen.put(eventId, true);
      }
    }
  }

  private boolean getBoolFromObject(final Object input) {
    if (input != null) {
      return Boolean.parseBoolean(input.toString());
    }
    return false;
  }
}

File: frontEnd/lib/models/user_group.dart
import 'package:frontEnd/imports/groups_manager.dart';
import 'package:frontEnd/models/group_interface.dart';

class UserGroup implements GroupInterface {
  final String groupId;
  final String groupName;
  final String icon;
  final String lastActivity;
  final Map<String, bool> eventsUnseen;

  bool muted;

  UserGroup(
      {this.groupName,
      this.groupId,
      this.icon,
      this.lastActivity,
      this.muted,
      this.eventsUnseen});

  UserGroup.debug(this.groupId, this.groupName, this.icon, this.lastActivity,
      this.muted, this.eventsUnseen);

  factory UserGroup.fromJson(Map<String, dynamic> json, String groupId) {
    Map<String, bool> eventsUnseenMap = new Map<String, bool>();
    for (String eventId in json[GroupsManager.EVENTS_UNSEEN].keys) {
      eventsUnseenMap.putIfAbsent(
          eventId, () => json[GroupsManager.EVENTS_UNSEEN][eventId]);
    }
    return UserGroup(
        groupId: groupId,
        groupName: json[GroupsManager.GROUP_NAME],
        icon: json[GroupsManager.ICON],
        lastActivity: json[GroupsManager.LAST_ACTIVITY],
        muted: json[GroupsManager.MUTED],
        eventsUnseen: eventsUnseenMap);
  }

  Map asMap() {
    return {
      GroupsManager.GROUP_ID: groupId,
      GroupsManager.GROUP_NAME: groupName,
      GroupsManager.ICON: icon,
      GroupsManager.LAST_ACTIVITY: lastActivity,
      GroupsManager.MUTED: muted,
      GroupsManager.EVENTS_UNSEEN: eventsUnseen
    };
  }

  @override
  String toString() {
    return "GroupName: $groupName GroupIcon: "
        "$icon LastActivity: $lastActivity Muted: $muted EventsUnsen: $eventsUnseen";
  }

  @override
  String getGroupName() {
    return groupName;
  }
}
